syntax = "proto3";

package server;

service Server {
    rpc Greet(GreetRequest) returns (GreetResponse);
    rpc Client(ClientRequest) returns (ClientResponse);
}

message GreetRequest {
    string name = 1;
    string address = 2;
}

message GreetResponse {
    string message = 1;
}

message ClientRequest {
    string name = 1;
}

message ClientResponse {
    string address_name = 1;
}